Mathlib/Order/CompleteLattice.lean

```diff
@@ -1787,15 +1789,33 @@ theorem binary_relation_sInf_iff {α β : Type*} (s : Set (α → β → Prop))
 
 section CompleteLattice
 
-variable [Preorder α] [CompleteLattice β]
+variable {ι : Sort*} [Preorder α] [CompleteLattice β] {s : Set (α → β)} {f : ι → α → β}
+
+protected lemma Monotone.sSup (hs : ∀ f ∈ s, Monotone f) : Monotone (sSup s) :=
+  fun _ _ h ↦ iSup_mono fun f ↦ hs f f.2 h
+#align monotone_Sup_of_monotone Monotone.sSup
+
+protected lemma Monotone.sInf (hs : ∀ f ∈ s, Monotone f) : Monotone (sInf s) :=
+  fun _ _ h ↦ iInf_mono fun f ↦ hs f f.2 h
+#align monotone_Inf_of_monotone Monotone.sInf
+
+protected lemma Antitone.sSup (hs : ∀ f ∈ s, Antitone f) : Antitone (sSup s) :=
+  fun _ _ h ↦ iSup_mono fun f ↦ hs f f.2 h
+
+protected lemma Antitone.sInf (hs : ∀ f ∈ s, Antitone f) : Antitone (sInf s) :=
+  fun _ _ h ↦ iInf_mono fun f ↦ hs f f.2 h
 
-theorem monotone_sSup_of_monotone {s : Set (α → β)} (m_s : ∀ f ∈ s, Monotone f) :
-    Monotone (sSup s) := fun _ _ h => iSup_mono fun f => m_s f f.2 h
-#align monotone_Sup_of_monotone monotone_sSup_of_monotone
+@[deprecated (since := "2024-05-29")] alias monotone_sSup_of_monotone := Monotone.sSup
```

Comment:
Since dot notation isn't actually useful here, I'm not sure deprecation is really deserved.

Reply:
It is useful thanks to the new expected type dot notation. `.sSup _ : Monotone _` should* elaborate to `Monotone.sSup`

* seems like there currently is a bug due to `Monotone` being defined as a forall. Not sure it's known

Reply:
Let's not deprecate until the bug is reported then.

Reply:
https://github.com/leanprover/lean4/issues/4761